src/watchmaker/utils/imds/detect/providers/azure_provider.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+"""AWS Provider."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals, with_statement)
+
+import logging
+from os.path import exists
+
+import watchmaker.utils as utils
+from watchmaker.utils.imds.detect.providers.provider import AbstractProvider
+
+
+class AzureProvider(AbstractProvider):
+    """Concrete implementation of the Azure cloud provider."""
+
+    identifier = 'azure'
+
+    def __init__(self, logger=None):
+        self.logger = logger or logging.getLogger(__name__)
+        self.metadata_url = (
+            'http://169.254.169.254/metadata/instance?api-version=2021-02-01'
+        )
+        self.vendor_file = '/sys/class/dmi/id/sys_vendor'
+        self.headers = {'Metadata': 'true'}
+
+    def identify(self):
+        """Identify Azure using all the implemented options."""
+        self.logger.info('Try to identify DO')
```

Comment:
What is "DO"?

Reply:
It was in the code I started from and did not notice, apparently it is Digital Ocean, another cloud provider.